Lib/fontTools/otlLib/optimize/gpos.py

```diff
@@ -3,22 +3,42 @@
 from functools import reduce
 from itertools import chain
 from math import log2
+from textwrap import dedent
 from typing import DefaultDict, Dict, Iterable, List, Sequence, Tuple
 
 from fontTools.misc.intTools import bit_count, bit_indices
 from fontTools.ttLib import TTFont
 from fontTools.ttLib.tables import otBase, otTables
-
-# NOTE: activating this optimization via the environment variable is
-# experimental and may not be supported once an alternative mechanism
-# is in place. See: https://github.com/fonttools/fonttools/issues/2349
-GPOS_COMPACT_MODE_ENV_KEY = "FONTTOOLS_GPOS_COMPACT_MODE"
-GPOS_COMPACT_MODE_DEFAULT = "0"
+from fontTools.config import register_option
+
+MODE_OPTION = register_option(
+    name="otlLib.optimize.gpos.mode",
```

Comment:
Cosimo: prepend `fontTools.`, actually use `f'{__name__}.mode'` (but that only works if the option is defined in the module)

Reply:
@anthrotype I did this but I went with the simple option 1, so I'm not using `__name__`